life_expectancy/cleaning.py

```diff
@@ -0,0 +1,136 @@
+"""Module for cleaning life expectancy data."""
+import argparse
+import pandas as pd
+
+def main(region: str) -> None:
+    """Main function."""
+    full_data = _load_data('life_expectancy/data')
+    region_data = _clean_data(full_data, region)
+    _save_data_csv('life_expectancy/data', region_data, region)
+
+def _clean_data(data: pd.DataFrame, region: str) -> pd.DataFrame:
+    """ Clean the data and return only a given region.
+
+    Args:
+        region: the name of the region to clean
+    """
+    full_data = (
+        data
+        .pipe(_melt_table)
+        .pipe(_clean_year_value_cols)
+        .pipe(_clean_region_col)
+    )
+    region_data = _select_region(full_data, region)
+
+    return region_data
+
+def _load_data(data_path: str) -> pd.DataFrame:
+    """Load the data.
+
+    Returns:
+        data (pd.DataFrame): The data.
+    """
+    return pd.read_table(f'{data_path}/eu_life_expectancy_raw.tsv')
+
+def _melt_table(data: pd.DataFrame) -> pd.DataFrame:
+    """Melt the table to long format.
+    
+    Args:
+        data (pd.DataFrame): The data.
+        
+    Returns:
+        data (pd.DataFrame): The data in long format.
+    """
+
+    # clean column names
+    data[['unit', 'sex', 'age', 'region']] = data['unit,sex,age,geo\\time'].str.split(
+        ',', expand=True
+    )
+    data.drop('unit,sex,age,geo\\time', axis=1, inplace=True)
+
+    # melt table
+    return data.melt(
+        id_vars=['unit', 'sex', 'age', 'region'],
+        var_name='year', value_name='value'
+    )
+
+def _clean_year_value_cols(data: pd.DataFrame) -> pd.DataFrame:
+    """Clean the year and value columns, ensure they are of the correct type and
+        drop rows with missing values.
+
+    Args:
+        data (pd.DataFrame): The data.
+
+    Returns:
+        data (pd.DataFrame): The data with cleaned year and value columns.
+    """
+    # Ensure year is an integer
+    data['year'] = data['year'].str.extract(
+        r'(\d+)', expand=False
+        ).astype(int)
+
+    # Ensure value is a float
+    data['value'] = data['value'].str.extract(
+        r'(\d+.\d+)', expand=False
+        ).astype(float)
+
+    # drop rows with missing values
+    data.dropna(inplace=True)
+
+    return data
+
+def _clean_region_col(data: pd.DataFrame) -> pd.DataFrame:
+    """
+    Clean the region column of the data.
+
+    Args:
+        data: The data to clean.
+
+    Returns:
+        The cleaned data.
+    """
+    data['region'] = data['region'].apply(
+        lambda region: region.split('_')[0]
+        )
+
+    return data
+
+def _select_region(data: pd.DataFrame, region: str) -> pd.DataFrame:
+    """Select the region.
+
+    Args:
+        data (pd.DataFrame): The data.
+        region (str): The region to select.
+
+    Returns:
+        data (pd.DataFrame): The data with only the specified region.
```

Comment:
I think the type hints are not necessary here since you already mentioned them above

Reply:
It is redundant yes. it was the automatic generation of the docstring ehehe